### DISCUSSION TITLE
Filters or Faceted Search planned?

### DISCUSSION THREAD

**Discussion by Raidus** [lnx-search/lnx#128] [Q&A]
Hey,

are filters or faceted search planned? From tantivy project I can see faceted search is implemented.

Thank you!

**Comment by ChillFish8** [comment]
Hi! 

You should be already able to make use of facets within lnx using a `term` query.
Currently filtering is a little bit left to be desired as the current query system limits how specific you can be with querying and filtering, although a multi-query will get you some what of the way there.

In 0.10 the query system has been rewritten to hopefully give you almost 100% coverage of the things you can do in tantivy minus aggregations, so this should solve the issue when that comes.

**Reply by Sembiance** [reply]
I was also looking for an example JSON payload for facets as I also do not see it in the docs… did you ever get it working?

**Reply by ChillFish8** [reply]
Sorry for the delay, but you should be able to do this with either a term query or a normal query:

```json
{
  "query": [
     { "occur": "must", "normal": "my tantivy parser query here" },   <-- Filter
     { "occur": "must", "fuzzy": "My nice fuzzy search here" },   <--- User search
  ]
}
```

**Reply by draganjovanovich** [reply]
I got this error:
{'status': 422, 'data': 'no variant of enum QueryKind found in flattened data at line 1 column 139'}

**Reply by ChillFish8** [reply]
Can you send a copy of your full query payload?

**Reply by ChillFish8** [reply]
Ah sorry missed this, Try doing `{"normal": {"ctx": "genre:Action"}}` instead
